From 5fd5977c788ba99607193fa0e8d1d2ac2db7fdd1 Mon Sep 17 00:00:00 2001
From: Billy Tsai <billy_tsai@aspeedtech.com>
Date: Thu, 17 Jul 2025 14:54:13 +0800
Subject: [PATCH 01/10] ltpi: Lock SGPIO output only during serial output
 source modification
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The SGPIOS output lock should only be applied when modifying the serial
output source, rather than being set prior to SGPIOS enablement.
Applying the output lock prematurely affects all SGPIO serial outputs,
which is undesirable in our use case—some SGPIO pins are not associated
with LTPI, but are instead used to bypass the BMC pin to the PFR.

Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
Change-Id: I13aaf86758733c4cc439bbc5c75b83651f16e023
(cherry picked from commit c5573ebc9d806b9d1d3c894a5e01c7abcd511cf8)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 cmd/aspeed/ltpi/ltpi-v2.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/cmd/aspeed/ltpi/ltpi-v2.c b/cmd/aspeed/ltpi/ltpi-v2.c
index 55d64c124d..d247d1d204 100644
--- a/cmd/aspeed/ltpi/ltpi-v2.c
+++ b/cmd/aspeed/ltpi/ltpi-v2.c
@@ -865,11 +865,8 @@ static void ltpi_scm_sgpios_init(struct ltpi_priv *ltpi)
 	}
 	/* Check whether LTPI is initialized */
 	state = ltpi_get_link_mng_state(ltpi);
-	if (state != LTPI_LINK_MNG_ST_OP) {
+	if (state != LTPI_LINK_MNG_ST_OP)
 		ltpi_scm_sgpios_serial_out_source_sel(SELECT_FROM_CSR);
-		/* Avoid the SGPIOS output unstable value to SGPIOM */
-		ltpi_scm_sgpios_serial_out_lock();
-	}
 	/*
 	 * The meaning of SGPIO_PARALLEL_OUT_PROTECT will be inverted in A2,
 	 * so this bit setting should be removed when migrating to A2.
@@ -1069,6 +1066,8 @@ struct bootstage_t ltpi_init(struct rom_context *rom_ctx, uint32_t pin_strap, bo
 			ltpi1_hpm_set_pins();
 			ltpi_scm_init(ltpi1);
 		}
+		/* Avoid the SGPIOS output unstable value to SGPIOM */
+		ltpi_scm_sgpios_serial_out_lock();
 		ltpi_scm_sgpios_serial_out_source_sel(SELECT_FROM_PARALLEL_IN);
 		/* Unlock serial out to pass the LTPI Parallel in to SGPIOM */
 		ltpi_scm_sgpios_serial_out_unlock();
-- 
2.34.1

